### PR TITLE
CRYP-200: As a User, I want to edit an Ethereum wallet, so that I can change the wallet name

### PR DESCRIPTION
**Pull request by alexbara2000**
-Added frontend for wallet name edit
-added backend for eth

image

image

image

### CODE REVIEW

File: packages/api/src/gateways/edge-gateway/eth_edge_gateway_strategy.ts [hunk 1]
```diff
@@ -22,6 +23,11 @@ export class EthEdgeGatewayStrategy extends AbstractEdgeGatewayStrategy {
         return this.request<WalletWithBalance>(Method.POST, {}, path, req);
     }
 
+    async updateWallet(req: UpdateWalletRequest): Promise<WalletWithBalance> {
+        const path = `users/${req.userId}/wallets/${req.address}`;
+        return this.request<WalletWithBalance>(Method.PUT, {}, path, req);
+    }
```
Comment (jason-gerard) [inline packages/api/src/gateways/edge-gateway/eth_edge_gateway_strategy.ts:29]: You aren't passing any request here, this won't work

**Review by jason-gerard — COMMENTED** [review]
Please test your code before submitting it for review.

File: packages/api/src/gateways/edge-gateway/eth_edge_gateway_strategy.ts (hunk 1 above)
Comment (alexbara2000) [reply packages/api/src/gateways/edge-gateway/eth_edge_gateway_strategy.ts:29]: It works

Comment (alexbara2000) [reply packages/api/src/gateways/edge-gateway/eth_edge_gateway_strategy.ts:29]: I think you are referring to the the implementation for btc_edge that has null in the request which will not work. Btc implementation is another feature so I will do it in the other pull request..

File: packages/api/src/gateways/edge-gateway/btc_edge_gateway_strategy.ts [hunk 2]
```diff
@@ -22,6 +23,11 @@ export class BtcEdgeGatewayStrategy extends AbstractEdgeGatewayStrategy {
         return this.request<WalletWithBalance>(Method.POST, {}, path, req);
     }
 
+    async updateWallet(req: UpdateWalletRequest): Promise<WalletWithBalance> {
+        const path = `users/${req.userId}/wallets/${req.address}`;
+        return this.request<WalletWithBalance>(Method.PUT, {}, path, null);
```
Comment (jason-gerard) [inline packages/api/src/gateways/edge-gateway/btc_edge_gateway_strategy.ts:28]: ?

File: packages/api/src/gateways/edge-gateway/btc_edge_gateway_strategy.ts [hunk 3]
```diff
@@ -22,6 +23,11 @@ export class BtcEdgeGatewayStrategy extends AbstractEdgeGatewayStrategy {
...
+    }
```
Comment (Andre-Ibrahim) [inline packages/api/src/gateways/edge-gateway/btc_edge_gateway_strategy.ts:29]: Update null to req before merging so you dont forget in your next pr

File: packages/eth-edge/src/services/wallets.service.ts [hunk 4]
```diff
@@ -98,4 +99,17 @@ export class WalletsService {
 
         return new WalletBuilder().setWallet(wallet).setBalance(balance).build();
     }
+
+    async update(updateWalletReq: UpdateWalletRequest): Promise<WalletWithBalance> {
+        const [, balance] = await Promise.all([
+            this.walletRepository.update(
```
Comment (jason-gerard) [inline packages/eth-edge/src/services/wallets.service.ts:105]: This is fine but its better to use save() instead for the active object design pattern

**Review by jason-gerard — APPROVED** [review]

**Review by Andre-Ibrahim — APPROVED** [review]
Looks good just dont forget to fix the btc gateway before merging

File: packages/api/src/gateways/edge-gateway/btc_edge_gateway_strategy.ts (hunk 2 above)
Comment (alexbara2000) [reply packages/api/src/gateways/edge-gateway/btc_edge_gateway_strategy.ts:28]: To be done in the next PR

Comment (alexbara2000) [reply packages/api/src/gateways/edge-gateway/btc_edge_gateway_strategy.ts:28]: nvm I did it

File: packages/eth-edge/src/services/wallets.service.ts (hunk 4 above)
Comment (alexbara2000) [reply packages/eth-edge/src/services/wallets.service.ts:105]: You are right, using save instead of update is a well needed micro optimization

**Pull request merged**